src/reservoir_computing_prediction.py

```diff
@@ -0,0 +1,83 @@
+import torch
+import numpy as np
+from matplotlib import pyplot as plt
+import pandas as pd
+from sklearn.metrics import mean_squared_error
+import auto_esn.utils.dataset_loader as dl
+from auto_esn.datasets.df import MackeyGlass
+from auto_esn.esn.esn import GroupedDeepESN,DeepESN
+from auto_esn.esn.reservoir.util import NRMSELoss
+from auto_esn.esn.reservoir.activation import tanh
+
+
+def test_esn(data, test_size, name):
+    mse = []
+    number_of_points = np.arange(40, 300, 20)
+    number_of_layers = [1, 2, 3]
+
+    for layers in  number_of_layers:
+        mse = []
+        for points_number in number_of_points:
+            seidr = dl.loader_explicit(data, test_size=int(data.size * test_size))
+            X, X_test, y, y_test = seidr()
+            step = int(data.size * (1 - test_size) / points_number)
+            indices  = torch.arange(0, int(data.size * (1 - test_size)) - 1, step)
+            X = X[indices]
+            y = y[indices]
+            esn = DeepESN(num_layers=layers)
+            esn.fit(X, y)
+            output = esn(X_test)
+            plt.clf()
+            plt.plot(range(int(data.size * test_size)), output.view(-1).detach().numpy(), 'r',label='predicted')
+            plt.plot(range(int(data.size * test_size)), y_test.view(-1).detach().numpy(), 'b',label='original')
+            plt.legend()
+            plt.savefig(f'../output/reservoir/{name}_layers_{layers}_train_{points_number}')
+            current_mse = mean_squared_error(y_test, output)
```

Comment:
Are you sure this computes a correct result? Seems suspicious that the value for 40 seems the best in both cases, especially looking at the graph of the actual predictions, but maybe this is right.